### PR TITLE
[infrastructure] Add (S)CSS linting infrastructure

### PR DESCRIPTION
**Pull request by traviskaufman**
* Add Stylelint config
* Add BEM linting
* Add SCSS linting
* Fix all general lint errors
* Remove theme scss files as we won't be using them moving forward

Part of #4464
Relates to #4539

NOTE: Assigning everyone since this should get feedback from all core team members

### CODE REVIEW

File: .stylelintrc.yaml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    -
+      resolveNestedSelectors: true
+  selector-id-pattern: ^mdl?-.+
+  # Names are more semantic than numbers
+  font-weight-notation: named-where-possible
+  # :root is an important enough selector that it always stands on it's own. Also, there is no
+  # reason to make root a compound selector (?)
+  selector-root-no-composition: true
+  # http://www.paulirish.com/2010/the-protocol-relative-url/
+  function-url-no-scheme-relative: true
```
Comment (Garbee) [inline .stylelintrc.yaml]: We should actually enforce using HTTPS wherever possible, not relative URLs.

Comment (traviskaufman) [inline .stylelintrc.yaml]: I think that's what setting this to `true` does https://github.com/stylelint/stylelint/blob/master/src/rules/function-url-no-scheme-relative/README.md

Comment (Garbee) [inline .stylelintrc.yaml]: AH, OK. I was searching around and saw the initial issue. They were saying relative URLs there were good and should be enforced, so I thought this was doing that instead.

**Comment by Garbee** [comment]
LGTM

**Comment by rfru** [comment]
LGTM
Very nice!

What are the // postcss-bem-linter: comments used for? (besides ignoring)

**Comment by traviskaufman** [comment]
@rfru it tells postcss-bem-linter that the code contained in between them is a BEM component - https://github.com/postcss/postcss-bem-linter#defining-a-component

I put the `end` comments in to avoid potential issues with sass concatenating files together and having the linter throw false positives.

File: .stylelintrc.yaml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+rules:
+  # Follow best practices
+  font-family-name-quotes: always-where-recommended
+  # http://stackoverflow.com/questions/2168855/is-quoting-the-value-of-url-really-necessary
+  function-url-quotes: never
+  # https://www.w3.org/TR/selectors/#attribute-selectors
+  # http://stackoverflow.com/q/3851091
+  selector-attribute-quotes: always
+  # Single-quotes are a convention throughout our codebase
+  string-quotes: single
```
Comment (sgomes) [inline .stylelintrc.yaml]: We've been doing double quotes in CSS and single quotes in JS, but I suppose single quotes everywhere works too.

**Comment by sgomes** [comment]
LGTM, can't believe the range of options available!

File: .stylelintrc.yaml (hunk 2 above)
Comment (traviskaufman) [inline .stylelintrc.yaml]: Ah okay no problem. I can change this. I don't mind dbl-quotes anyway. They look like attributes and that's what's used in the WebIDL for attrs (AFAIK).

**Pull request merged**